FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift

```diff
@@ -169,6 +170,25 @@ extension EncryptedStorage {
         }
     }
 
+    func updateKeys(keyDetails: [KeyDetails], passPhrase: String, source: KeySource) {
+        // KeyInfo doesn't have primaty key, to avoid migration we need to delete keys and then save them
+
+        // delete keys
+        keyDetails.forEach { keyDetail in
+            try? storage.write {
+                storage.delete(storage.objects(KeyInfo.self)
+                    .filter("longid=%@", keyDetail.ids[0].longid))
```

Comment:
We should start saving fingerprints in the storage (in addition to longids), and compare by fingerprints wherever possible instead of longids. It's safer - collisions less likely. I'll make another issue for this.